source/Core/Threads/GUIThread.cpp

```diff
@@ -996,6 +1000,43 @@ void startGUITask(void const *argument) {
   }
 #endif
 #endif
+  // Calibrate Cold Junction Compensation directly at boot, before internal components get warm.
+  while (preStartChecks() == 0) {
+    ulTaskNotifyTake(pdTRUE, TICKS_100MS);
+  }
```

Comment:
@Ralim
Even after putting in what you suggested it starts without the tip being attached.

Most likely I did do it the wrong way! 🤷‍♂️

Reply:
>I would yank this block of logic you have here out into a function like performTipOffsetCalibration()..


I just meant to take the blob of code have added here and make it into its own function.
(yank just being slang for moving it up the file).

So i would have roughly:

```C
void performCJC(){
 while (preStartChecks() == 0) {
    ulTaskNotifyTake(pdTRUE, TICKS_100MS);
  }
 // Insert the rest of your code section here
}
```

Then where you currently have you code I would just do a 
```C

if (setting Is Set) {
performCJC();
}
```

Reply:
I tried that but I was not able to get it working as I kept getting 2 errors no matter what I changed.

Reply:
I tried that but I was not able to get it to work as I kept getting 2 errors no matter what I changed.

Reply:
Ok, that worked!
But still neither the ***missing tip condition*** nor the ***temperature condition*** are respected.

Reply:
```C#
if (!isTipDisconnected() && Temp <= 32) {
```
As expected this did not solve the 'condition-issue'.

Do you have any Idea what could be the cause?

Reply:
@Ralim
```C#
if (!isTipDisconnected() && Temp <= 32) {
```
As expected this did not solve the 'condition-issue'.

Do you have any Idea what could be the cause?

Reply:
Ok, that worked!
But still neither the ***missing tip condition*** nor the ***temperature condition*** are respected.

@Ralim
```C#
if (!isTipDisconnected() && Temp <= 32) {
```
As expected this did not solve the 'condition-issue'.

Do you have any Idea what could be the cause?

Maybe some **function** to *measure the temperature* is not loaded at this point in time?